src/jit/lsra.h

```diff
@@ -847,11 +847,17 @@ class LinearScan : public LinearScanInterface
     };
 
 private:
+    bool enregisterLocalVars()
+    {
+        return localVarIntervals != nullptr;
```

Comment:
I think I'd be more comfortable just using a boolean field that is set in the constructor for this. The code in `identifyCandidates` in particular reads pretty strangely as it must set `localVarIntervals` before calling this method (and I would have expected it to use this method to decide whether or not to allocate `localVarIntervals` in the first place). To be concrete, something like:

```
private:
    bool m_enregisterLocalVars;
    bool enregisterLocalVars() const
    {
        return m_enregisterLocalVars;
    }

     ...

    LinearScan()
    {
        m_enregisterLocalVars = ((compiler->opts.compFlags & CLFLG_REGVAR) == 0) && compiler->lvaTrackedCount > 0;
    }
```